runtime/plaid/src/apis/aws/dynamodb.rs

```
@@ -0,0 +1,538 @@
+    write: HashMap<String, HashSet<String>>,
+    /// Configured readers - maps a table name to a list of rules that are allowed to read it
+    read: HashMap<String, HashSet<String>>,
+}
+
+/// Represents the KMS API that handles all requests to KMS
+pub struct DynamoDb {
+    /// The underlying KMS client used to interact with the KMS API.
+    client: Client,
+    /// Configured writers - maps a table name to a list of rules that are allowed to write it
+    write: HashMap<String, HashSet<String>>,
+    /// Configured readers - maps a table name to a list of rules that are allowed to read it
+    read: HashMap<String, HashSet<String>>,
+}
+
+impl DynamoDb {
+    /// Creates a new instance of `Kms`
+    pub async fn new(config: DynamoDbConfig) -> Self {
+        let DynamoDbConfig {
+            authentication,
+            write,
+            read,
+        } = config;
+        let sdk_config = get_aws_sdk_config(authentication).await;
+        let client = Client::new(&sdk_config);
+
+        Self {
+            client,
+            write,
+            read,
+        }
+    }
+
+    // PutItem function with direct deserialization
+    pub async fn put_item(
+        &self,
+        module: Arc<PlaidModule>,
+        table_name: &str,
+        item_json: Value,
+        expression_attribute_names_json: Option<Value>,
+        expression_attribute_values_json: Option<Value>,
+        condition_expression: Option<&str>,
+        return_values: Option<&str>,
+    ) -> Result<Option<Value>, String> {
+        if let Some(write_access) = self.write.get(table_name) {
+            // check if this module has write access to this table
+            if !write_access.contains(&module.to_string()) {
+                return Err(format!("{module} no write access for {table_name}"));
+            }
+        } else {
+            return Err(format!("{module} no write access for {table_name}"));
```

Maybe we can return a different error, like "unknown table".